server/server.js

```diff
@@ -23,13 +61,42 @@ app.use(passport.session());
 
 /* Routes */
 app.use('/api/user', userRouter);
+app.use('/api/aws', awsRouter);
+app.use('/api/message', messageRouter);
+app.use('/api/vendor', vendorRouter);
+app.use('/api/search', searchRouter);
+app.use('/api/specialfeatures', specialFeaturesRouter);
+app.use('/api/vendortypes', vendorTypesRouter);
+
 
 // Serve static files
 app.use(express.static('build'));
 
 // App Set //
 const PORT = process.env.PORT || 5000;
 
+//Whenever someone connects this gets executed
+io.on('connection', (socket) => {
+  console.log('a user connected!!!!!!!');
+
+  socket.on('disconnect', () => {
+    console.log('User has disconnected!');
+  });
+
+  socket.on('join', ({ name, room }) => {
+    console.log('User has joined chat!', name, room);
+  });
+
+  socket.on('send message', (message) => {
+    console.log('Got a message', message);
+    io.emit('send message', message);
+  });
+});
+
+http.listen(4000, function () {
```

Comment:
This isn't quite right -- you shouldn't need two port listeners here. This may give you trouble on heroku.

Check out: https://socket.io/get-started/chat
It looks similar to what you have here, except without this `http.listen()` call.

Honestly, I've never done socket.io before, so if this is what you managed to get work 👍  We can always cross the heroku bridge when we come to it.